packages/features/settings/outOfOffice/CreateOrEditOutOfOfficeModal.tsx

```diff
@@ -43,21 +48,39 @@ export const CreateOrEditOutOfOfficeEntryModal = ({
   const { t } = useLocale();
   const utils = trpc.useUtils();
 
-  const { data: listMembers } = trpc.viewer.teams.legacyListMembers.useQuery({});
+  const [searchText, setSearchText] = useState("");
+  const debouncedSearch = useDebounce(searchText, 500);
+
+  const members = trpc.viewer.teams.legacyListMembers.useInfiniteQuery(
+    { limit: 10, searchText: debouncedSearch },
+    {
+      enabled: true,
+      getNextPageParam: (lastPage) => lastPage.nextCursor,
+    }
+  );
 
   const me = useMeQuery();
 
   const memberListOptions: {
     value: number;
     label: string;
+    avatarUrl: string | null;
   }[] =
-    listMembers
+    members?.data?.pages
+      .flatMap((page) => page.members)
       ?.filter((member) => me?.data?.id !== member.id)
       .map((member) => ({
         value: member.id,
         label: member.name || "",
```

Comment:
I think we need to fall back to email here because when no names are set you get this weird state in the OOO list
![image](https://github.com/user-attachments/assets/9a03d116-e09f-40e3-b10a-cef151140f6f)

Reply:
Fixed here: https://github.com/calcom/cal.com/pull/17396/commits/c9a45137bb09027e803c78e32b2e597bad7a5541 fall back to username instead